### PR TITLE
chore: move Loadable to kit [WEB-1688]

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Description

- Moves Loadable to UI Kit `utils` folder and updates imports
- Refactor Loadable to use methods (from @ClaireNeveu)
- Ensure that UI Kit components taking Loadables as props (currently only `CodeEditor`, `LineChart`, and `Spinner`) make them optional
- Minor relating refactoring

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/components/kit/utils/loadable.ts [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+  | {
+      Loaded: (data: T) => U;
+      _: () => U;
+    }
+  | {
+      NotLoaded: () => U;
+      _: () => U;
+    };
+
+class Loadable_<T> {
```
Comment (ClaireNeveu) [inline webui/react/src/components/kit/utils/loadable.ts:18]: You don't have to do it in this PR but can we at least make a ticket to add tests for loadable?

Comment (EmilyBonar) [reply webui/react/src/components/kit/utils/loadable.ts:18]: Done, WEB-1711

File: webui/react/src/components/kit/Spinner.tsx [hunk 2]
```diff
@@ -12,20 +13,20 @@ interface PropsBase {
...
-      spinning?: boolean;
-    }
-  | {
-      children: (data: T) => JSX.Element;
-      conditionalRender?: never;
-      data: Loadable<T>;
-      spinning?: never;
-    }
-) &
+type Props<T> = XOR<
```
Comment (ClaireNeveu) [inline webui/react/src/components/kit/Spinner.tsx:16]: I don't realize understand what the `XOR` type does; it's not documented.

**Review by ashtonG — APPROVED** [review]

File: webui/react/src/components/kit/Spinner.tsx (hunk 2 above)
Comment (EmilyBonar) [reply webui/react/src/components/kit/Spinner.tsx:16]: It's a utility type that allows you to create mutually exclusive types where either the first or second set of props must be present, but not both. I got it from [here](https://stackoverflow.com/a/53229857). Basically, it does the same thing as the union that was there previously, but more cleanly. Of course, I forgot to actually finish cleaning it up until now, but it means that the props can look like this, without any `never`s in the mix:
```
type Props<T> = XOR<
  {
    children?: React.ReactNode;
    conditionalRender?: boolean;
    spinning?: boolean;
  },
  {
    children: (data: T) => JSX.Element;
    data: Loadable<T>;
  }
> &
  PropsBase;
```

Comment (EmilyBonar) [reply webui/react/src/components/kit/Spinner.tsx:16]: I added a link to the StackExchange answer next to the XOR definition.

Comment (ClaireNeveu) [reply webui/react/src/components/kit/Spinner.tsx:16]: Makes sense. With the `never`s removed the function is clear.

**Pull request merged**